pygmt/helpers/utils.py

```diff
@@ -205,7 +205,55 @@ def data_kind(data=None, x=None, y=None, z=None, required_z=False, required_data
     return kind
 
 
-def non_ascii_to_octal(argstr: str) -> str:
+def check_encoding(argstr: str) -> str:
+    """
+    Check the charset encoding of a string.
+
+    All characters in the string must be in the same charset encoding, otherwise the
+    default ISOLatin1+ encoding is returned. Characters in the Symbol and ZapfDingbats
+    fonts are also checked because they're independent on the setting of the charset.
+
+    Parameters
+    ----------
+    argstr
+        The string to be checked.
+
+    Returns
+    -------
+    encoding
+        The encoding of the string.
+
+    Examples
+    --------
+    >>> check_encoding("123ABC+-?!")  # ASCII characters only
+    'ISOLatin1+'
+    >>> check_encoding("12AB±β①②")  # Characters in ISOLatin1+
+    'ISOLatin1+'
+    >>> check_encoding("12ABāáâãäåβ①②")  # Characters in ISO-8859-4
+    'ISO-8859-4'
+    >>> check_encoding("12ABŒā")  # Mix characters in ISOLatin1+ (Œ) and ISO-8859-4 (ā)
+    'ISOLatin1+'
+    >>> check_encoding("123AB中文")  # Characters not in any charset encoding
+    'ISOLatin1+'
+    """
```

Comment:
In addition to the GMT-supported encodings, I'm thinking if we should return `ascii` (the name `ascii` comes from https://docs.python.org/3/library/codecs.html#standard-encodings) if the input string only contains ASCII characters, e.g.,
```
    if all(32 <= ord(c) <= 126 for c in argstr):
        return "ascii"
```
Since in most cases, the arguments and the text string contain ASCII characters only. When `ascii` encoding is detected, we no longer need to apply `non_ascii_to_octal` to the strings, which may improve the performance for most cases.

Reply:
Done in 2d01f6b8fecb3725bffe8e51531629464bbf7f90.